packages/storage-plus/src/map.rs

```diff
@@ -108,6 +110,25 @@ where
     }
 }
 
+#[cfg(feature = "iterator")]
+impl<'a, K, T> Map<'a, K, T>
+where
+    T: Serialize + DeserializeOwned,
+    K: PrimaryKey<'a> + Deserializable,
```

Comment:
It is like that already (`K::Output` is `Deserializable`). I tried to avoid requiring `K` to be `Deserializable`, but hit some compiler errors when not using `K` as output for the full key. Will revisit, to see if I can improve on it.

Reply:
This requires `K` to be `Deserializable` just to be able to infer the `Output` type.

Let's revisit your idea of a `K::Owned` trait attribute in another iteration.